it/src/test/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManagerTest.java

```diff
@@ -0,0 +1,236 @@
+    testManager =
+        spy(
+            new DefaultPubsubResourceManager(
+                TEST_ID, PROJECT_ID, null, topicAdminClient, subscriptionAdminClient));
+
+    when(topic.getName()).thenReturn("projects/" + PROJECT_ID + "/topics/" + TOPIC_NAME);
+    when(subscription.getName())
+        .thenReturn("projects/" + PROJECT_ID + "/subscriptions/" + SUBSCRIPTION_NAME);
+    doReturn(publisher).when(testManager).getPublisher(any(TopicName.class));
+  }
+
+  @Test
+  public void testBuilderWithInvalidProject() {
+    IllegalArgumentException exception =
+        Assert.assertThrows(
+            IllegalArgumentException.class, () -> DefaultPubsubResourceManager.builder(""));
+    assertThat(exception).hasMessageThat().contains("projectId can not be empty");
+  }
+
+  @Test
+  public void testInvalidTopicName() {
+    IllegalArgumentException exception =
+        Assert.assertThrows(IllegalArgumentException.class, () -> testManager.getTopicName(""));
+    assertThat(exception).hasMessageThat().contains("topicName can not be empty");
+  }
+
+  @Test
+  public void testValidTopicName() {
+    TopicName topicName = testManager.getTopicName("valid-topic");
+    assertThat(topicName).isEqualTo(TopicName.of(PROJECT_ID, "test-id-valid-topic"));
+  }
+
+  @Test
+  public void testInvalidSubscriptionName() {
+    IllegalArgumentException exception =
+        Assert.assertThrows(
+            IllegalArgumentException.class, () -> testManager.getSubscriptionName(""));
+    assertThat(exception).hasMessageThat().contains("subscriptionName can not be empty");
+  }
+
+  @Test
+  public void testValidSubscriptionName() {
+    SubscriptionName subscriptionName = testManager.getSubscriptionName("valid-sub");
+    assertThat(subscriptionName).isEqualTo(SubscriptionName.of(PROJECT_ID, "test-id-valid-sub"));
+  }
+
+  @Test
+  public void testCreateTopic() {
+    // Arrange
+    when(topicAdminClient.createTopic(any(TopicName.class))).thenReturn(topic);
+
+    // Act
+    TopicName createTopic = testManager.createTopic("topic-name");
+
+    // Assert
+    assertThat(createTopic).isNotNull();
+    verify(topicAdminClient).createTopic(topicNameCaptor.capture());
+    TopicName actualTopicName = topicNameCaptor.getValue();
+    assertThat(actualTopicName)
+        .isEqualTo(TopicName.of(PROJECT_ID, String.format("%s%s", TEST_ID, "topic-name")));
+  }
+
+  @Test
+  public void testCreateSubscription() {
+    // Arrange
+    when(subscriptionAdminClient.createSubscription(
+            any(SubscriptionName.class), any(TopicName.class), any(PushConfig.class), anyInt()))
+        .thenReturn(subscription);
+
+    // Act
+    SubscriptionName createSub = testManager.createSubscription("topic-name", "subscription-name");
+
+    // Assert
+    assertThat(createSub).isNotNull();
+    verify(subscriptionAdminClient)
+        .createSubscription(
+            subscriptionNameCaptor.capture(),
+            topicNameCaptor.capture(),
+            any(PushConfig.class),
```

Comment:
Just `any()` is good enough. (I'm not sure specifying the class offers any value here.)

Same goes for things like `anyInt()`.

Reply:
Thanks, will use `any()` for non-overloaded methods.
However, `any()` breaks when used in lieu of primitives.